file_catalog/server.py

```diff
@@ -227,16 +229,52 @@ def get(self):
     @coroutine
     def post(self):
         metadata = json_decode(self.request.body)
-        ret = yield self.db.get_file({'file_name':metadata['file_name']})
+
+        # check metadata for mandatory fields
+        if not all(x in metadata for x in ['locations', 'uid', 'checksum']):
+            # locations (url) is mandatory
+            self.send_error(400, message='mandatory metadata missing',
+                            file=self.files_url)
+            return
+        elif any(x in metadata for x in ['_id', 'mongo_id']):
+            # forbidden fields
+            self.send_error(400, message='forbidden attributes',
+                            file=self.files_url)
+            return
+        elif not isinstance(metadata['locations'], list):
+            # locations needs to be a list
+            self.send_error(400, message='member `locations` must be a list',
+                            file=self.files_url)
+            return
+        elif not len(metadata['locations']):
+            # location needs have at least one entry
+            self.send_error(400, message='member `locations` must be a list with at least one url',
+                            file=self.files_url)
+            return
+        elif any(not len(l) for l in metadata['locations']):
```

Comment:
Similar to above, better as:

```python
elif not all(l for l in metadata['locations'])
```

Actually, now that I think about it you could remove the previous check.  This one would catch that condition as well.